core/src/main/java/tachiyomi/core/util/system/ImageUtil.kt

```diff
@@ -625,6 +709,33 @@ object ImageUtil {
         return options
     }
 
+    // SY -->
+    private fun extractImageOptionsZip4j(zip4jFile: ZipFile?, zip4jEntry: FileHeader?): BitmapFactory.Options {
+        zip4jFile?.getInputStream(zip4jEntry).use { imageStream ->
+            val imageBytes = imageStream?.readBytes()
+            val options = BitmapFactory.Options().apply { inJustDecodeBounds = true }
+            imageBytes?.size?.let { BitmapFactory.decodeByteArray(imageBytes, 0, it, options) }
```

Comment:
This might not work with image formats that by default Android doesn't support. Tachiyomi usually uses `ImageDecoder.newInstance(imageBytes.inputStream())?.decode()!!` and the like to decode images.

Reply:
This should probably be changed upstream since I did not change the decoder that is being used [here](https://github.com/jobobby04/TachiyomiSY/blob/8ad63865bbc9fd4249ecdfb7d1c3c24915ce8456/core/src/main/java/tachiyomi/core/util/system/ImageUtil.kt#L703-L709) or any of the functions code that does not pertain to getting the input-stream for that matter. This function only exists because zip4j input streams do not support the `mark()` and `reset()` functions used in `extractImageOptions()` and thus require a fresh input stream every time. 

I would also **guess** that using `BitmapFactory.decodeByteArray` would be necessary to get  `BitmapFactory.Options()` which this function returns, but that could be wrong and I will gladly change the decoder to `ImageDecoder.newInstance` if it is usable and you want me to change it here instead of upstream. 

Thank you very much for the code reviews

Reply:
Got it, it should be fine then, I checked and I see main Tachi doing the same thing 👍